portal/src/main/webapp/js/src/oncoprint/setup.js

```diff
@@ -1576,19 +1607,28 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
 		    QuerySession.getAlteredSamples(),
 		    QuerySession.getAlteredPatients(),
 		    QuerySession.getCaseUIDMap(),
+		    QuerySession.getStudyPatientMap(),
 		    dataPopulatedPromise)
 	    .then(function (patient_ids,
 		    altered_samples,
 		    altered_patients,
-		    case_uid_map) {
+		    case_uid_map,
+		    studyPatientMap) {
 		console.log("in initOncoprint, altered cases fetched, setting zoom level");
-		if ((State.using_sample_data ? QuerySession.getSampleIds() : patient_ids).length > 200) {
-		    // TODO: assume multiple studies
-		    var study_id = QuerySession.getCancerStudyIds()[0];
-		    var getUID = function(id) {
-			return case_uid_map[study_id][id];
+		var _data = (State.using_sample_data ? QuerySession.getStudySampleMap() : studyPatientMap);
+		if (Object.keys(_data).length > 200) {
+			var getUID = function (studyIdsList,caseId) {
+		    	var tempUIDs = [];
+		    	studyIdsList.forEach(function(studyId){
+		    		tempUIDs.push(case_uid_map[studyId][caseId])
+		    	})
+		    	return tempUIDs;
 		    };
-		    oncoprint.setHorzZoomToFit(State.using_sample_data ? altered_samples.map(getUID) : altered_patients.map(getUID));
+		    var _resultcaseStudyCaseMap = {};
+		    $.each((State.using_sample_data ? altered_samples : altered_patients), function(key,caseId){
```

Comment:
you should use altered_sample_uids and altered_patient_uids https://github.com/kalletlak/cbioportal/blob/cd90cf5651005065ee181ffcc3f5dfeeb56159e3/portal/src/main/webapp/js/api/cbioportal-datamanager.js#L1634